src/main/java/bio/terra/service/snapshotbuilder/query/filtervariable/BinaryFilterVariable.java

```diff
@@ -38,16 +38,16 @@ public enum BinaryOperator implements SqlExpression {
     }
 
     @Override
-    public String renderSQL(CloudPlatformWrapper platform) {
+    public String renderSQL(SqlRenderContext context) {
       return sql;
     }
   }
 
-  public static BinaryFilterVariable equals(FieldVariable fieldVariable, Literal value) {
+  public static BinaryFilterVariable equals(FieldVariable fieldVariable, SqlExpression value) {
```

Comment:
This isn't related to the refactor, it's from my other branch where this change started. Instead of only allowing comparing against literals, this now allows comparing against other expressions including other fields. Normally this is done via joins but in some cases we need to filter on a comparison that's not a join.